java/processor-functions/src/main/java/io/deephaven/processor/functions/ChunkUtils.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.processor.functions;
+
+import io.deephaven.chunk.ObjectChunk;
+import io.deephaven.chunk.WritableBooleanChunk;
+import io.deephaven.chunk.WritableByteChunk;
+import io.deephaven.chunk.WritableCharChunk;
+import io.deephaven.chunk.WritableDoubleChunk;
+import io.deephaven.chunk.WritableFloatChunk;
+import io.deephaven.chunk.WritableIntChunk;
+import io.deephaven.chunk.WritableLongChunk;
+import io.deephaven.chunk.WritableObjectChunk;
+import io.deephaven.chunk.WritableShortChunk;
+import io.deephaven.functions.ToByteFunction;
+import io.deephaven.functions.ToCharFunction;
+import io.deephaven.functions.ToFloatFunction;
+import io.deephaven.functions.ToShortFunction;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.function.ToDoubleFunction;
+import java.util.function.ToIntFunction;
+import java.util.function.ToLongFunction;
+
+final class ChunkUtils {
+
+    // Copy from io.deephaven.kafka.ingest.ChunkUtils
```

Comment:
Why are we copying this? Seems like we could just move it somewhere better.

Reply:
It's possible that these should move into code generation of Chunks themselves, but trying to grok that code generation logic and deal with generic typing is a pita and a lot more work.

I'm hoping the kafka ChunkUtils eventually gets deleted, and I don't know how much value there is to making this publicly accessible otherwise until necessary.

Reply:
I was able to remove the duplication and remove a lot of code by creating `FieldCopierProcessorImpl`